File: javascript/etl/etl.js
var ETL = function(){};

ETL.prototype.transform = function(oldCode){
  refactoredCode = {};
  for (var key in oldCode){
    for(var i = 0; i < oldCode[key].length; i++ ){
      refactoredCode[oldCode[key][i].toLowerCase()] = parseInt(key);
    }
  }
  return refactoredCode;
};

module.exports = ETL;

File: java/etl/src/main/java/Etl.java
import java.util.HashMap;
import java.util.List;
import java.util.Map;

public class Etl {

    public Map<String, Integer> transform(Map<Integer, List<String>> old) {
        Map<String, Integer> transformedMap = new HashMap<String, Integer>();

        old.forEach((k,v) -> {
            v.forEach((letter) -> {
                transformedMap.put(letter.toLowerCase(), k);
            });
        });

        return transformedMap;
    }

}
